shared/vcjs/verifyCredential.ts

```diff
@@ -21,41 +21,81 @@ const ProofPurpose = {
 
 export async function verifyCredential(
   verifiableCredential: VerifiableCredential | Credential,
-): Promise<boolean> {
-  let purpose: PublicKeyProofPurpose | AssertionProofPurpose;
-  switch (verifiableCredential.proof.proofPurpose) {
-    case ProofPurpose.PublicKey:
-      purpose = new PublicKeyProofPurpose();
-      break;
-    case ProofPurpose.Assertion:
-      purpose = new AssertionProofPurpose();
-      break;
-  }
-
-  let suite: Ed25519Signature2018 | RsaSignature2018;
-  const suiteOptions = {
-    verificationMethod: verifiableCredential.proof.verificationMethod,
-    date: verifiableCredential.proof.created,
-  };
-  switch (verifiableCredential.proof.type) {
-    case ProofType.ED25519: {
-      suite = new Ed25519Signature2018(suiteOptions);
-      break;
+): Promise<VerificationResult> {
+  try {
+    let purpose: PublicKeyProofPurpose | AssertionProofPurpose;
+    switch (verifiableCredential.proof.proofPurpose) {
+      case ProofPurpose.PublicKey:
+        purpose = new PublicKeyProofPurpose();
+        break;
+      case ProofPurpose.Assertion:
+        purpose = new AssertionProofPurpose();
+        break;
     }
-    case ProofType.RSA: {
-      suite = new RsaSignature2018(suiteOptions);
-      break;
+
+    let suite: Ed25519Signature2018 | RsaSignature2018;
+    const suiteOptions = {
+      verificationMethod: verifiableCredential.proof.verificationMethod,
+      date: verifiableCredential.proof.created,
+    };
+    switch (verifiableCredential.proof.type) {
+      case ProofType.ED25519: {
+        suite = new Ed25519Signature2018(suiteOptions);
+        break;
+      }
+      case ProofType.RSA: {
+        suite = new RsaSignature2018(suiteOptions);
+        break;
+      }
     }
+
+    const vcjsOptions = {
+      purpose,
+      suite,
+      credential: verifiableCredential,
+      documentLoader: jsonld.documentLoaders.xhr(),
+    };
+
+    const result = await vcjs.verifyCredential(vcjsOptions);
+    return handleError(result, false);
```

Comment:
do we need to have isException param, as we know if the result is null then it is a exception in all other cases we object result as an object.